EXTENSIONS.md

```diff
@@ -0,0 +1,173 @@
+# Extensions
+
+Podman Desktop is organized so that you can modularly add new functionality in the form of "extensions" as well as the corresponding `extension-api`. This allows you to communicate with Podman Desktop without having to know the internal-workings. You look for the API call and Podman Desktop will do the rest.
+
+Bundled extensions are located in the `/extensions` folder.
+
+#### Creating a new extension
+
+When creating a new extension, import the extension API: `import * as extensionApi from '@podman-desktop/api';` 
+
+All functionality with Podman Desktop is communicated through this API including registering the new extension. The API is located [here](https://github.com/containers/podman-desktop/blob/main/packages/extension-api/src/extension-api.d.ts).
+
+When loading an extension, Podman Desktop will:
+1. Search and load the JavaScript file specified in `main` entry of the `package.json` file in the extension directory (typically `extension.js`).
+2. Run the exported `activate` function.
+
+When unloading an extension, Podman Desktop will:
+1. Run the exported `deactivate` function.
+
+#### Example boilerplate code
+
+This is an example `extensions/foobar/src/extensions.ts` file with the basic `activate ` and `deactivate` functionality:
+
+```ts
+import * as extensionApi from '@podman-desktop/api';
+
+
+// Activate the extension asynchronously
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+
+  // Create a provider with an example name, ID and icon
+  const provider = extensionApi.provider.createProvider({
+    name: 'foobar',
+    id: 'foobar',
+    status: 'unknown',
+    images: {
+      icon: './icon.png',
+      logo: './icon.png',
+    },
+  });
+
+  // Push the new provider to Podman Desktop
```

Comment:
The subscriptions array is for adding disposable resources. It provides a way to free resources that not needed after the extension is deactivated. All disposable resources are going to be disposed (dispose method called) when extension is deactivated.